lib/publiccloud/azure.pm

```diff
@@ -639,7 +639,7 @@ sub cleanup {
 
     if (!check_var('PUBLIC_CLOUD_SLES4SAP', 1) && defined($args->{my_instance}->{instance_id})) {
         my $id = $args->{my_instance}->{instance_id};
-        script_run("az vm boot-diagnostics get-boot-log --ids $id | jq -r '.' > bootlog.txt", timeout => 120, die_on_timeout => 0);
+        script_run("timeout 110 az vm boot-diagnostics get-boot-log --ids $id | jq -r '.' > bootlog.txt", timeout => 120);
```

Comment:
this is disruptive change , with current master after waiting 2 minutes in case command hangs test will not die and keep running further as nothing have happened and in new version this will not happen so upload of bootlog will not happen and call to `$self->SUPER::cleanup();` will not happen either . 

In this case `die_on_timeout => 0` is only possible solution but if you saying that option will be dropped no matter what I will say then I would at least wrap it `eval` to allow code keep working as it worked before . In any case like I said in my initial message this will require additional testing and analysis

Reply:
The `timeout 110 …` would ensure that the command is cancelled after 110s leaving 10s to terminate the command before the timeout of `script_run` would be reached. If you say that the problem is that also that is not enough we can also do the same as I did in other cases in this PR which would be

```suggestion
       enter_cmd "az vm boot-diagnostics get-boot-log --ids $id | jq -r '.' > bootlog.txt; echo DONE > /dev/$serialdev";
       wait_serial 'DONE', timeout => 120;
```

Reply:
I don't understand why `die_on_timeout` is being deprecated and I'd like to learn more about it.

If we decide to proceed I suggest the following as less complicated approach:
```suggestion
        script_run("timeout 110 az vm boot-diagnostics get-boot-log --ids $id | jq -r '.' > bootlog.txt; true", timeout => 120);
```

Reply:
The behaviour of `die_on_timeout` is the default since years and that will stay. Only the possibility to set to `0` will be gone as per discussion in
* #1807

https://github.com/os-autoinst/os-autoinst/pull/1807#issuecomment-946107048 is where @cfconrad lays out why this behaviour should be deprecated which actually originally suggested to introduce this as a breaking change without a way to still keep the old behaviour. This was supported in https://github.com/os-autoinst/os-autoinst/pull/1807#issuecomment-975578291 by @Vogtinator as well as in https://github.com/os-autoinst/os-autoinst/pull/1807#issuecomment-975959097 by @jlausuch . In https://github.com/os-autoinst/os-autoinst/pull/1807#issuecomment-948882770 I suggested to only deprecate the old behaviour, later change the default - which we did, and then finally remove the old deprecated behaviour which we now plan to do after tests had been riddled with the deprecation warning for years and were ignored :)

Reply:
> The behaviour of `die_on_timeout` is the default since years and that will stay. Only the possibility to set to `0` will be gone as per discussion in
> 
> * [New test: Seahorse #1807](https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/1807)
> 
> [os-autoinst/os-autoinst#1807 (comment)](https://github.com/os-autoinst/os-autoinst/pull/1807#issuecomment-946107048) is where @cfconrad lays out why this behaviour should be deprecated which actually originally suggested to introduce this as a breaking change without a way to still keep the old behaviour. This was supported in [os-autoinst/os-autoinst#1807 (comment)](https://github.com/os-autoinst/os-autoinst/pull/1807#issuecomment-975578291) by @Vogtinator as well as in [os-autoinst/os-autoinst#1807 (comment)](https://github.com/os-autoinst/os-autoinst/pull/1807#issuecomment-975959097) by @jlausuch . In [os-autoinst/os-autoinst#1807 (comment)](https://github.com/os-autoinst/os-autoinst/pull/1807#issuecomment-948882770) I suggested to only deprecate the old behaviour, later change the default - which we did, and then finally remove the old deprecated behaviour which we now plan to do after tests had been riddled with the deprecation warning for years and were ignored :)

I have read all comments which you mention . I see that @cfconrad  is just stating that in his opinion `die_on_timeout => 0` is misuse of `script_run` without actually explaining WHY it is misusing ( which is not obvious to me at least ) . And @Vogtinator and @jlausuch  comments are not making this part more clear but more discussing around what would happen if we not explicitly disable this ability . So at least for me it is not clear WHY this deprecation is necessary in first place . 
BUT I also must highlight the fact the I have seen deprecation warnings and was not raising this discussion earlier so I probably already lost my window of abilities to raise this topic so I will obey your changes and switch from  `script_run` to `enter_cmd`
BUT  I also must insist that just deletion of `die_on_timeout => 0` and replacing it with timeout within a bash command won't help in our case and we really need to migrate to `enter_cmd` in such a case .
The best solution for me time would be revert of changes in terms of this big PR and allowing us to do it internally ( I promise to do this ASAP to not block you with your changes )

Reply:
> The best solution for me time would be revert of changes in terms of this big PR and allowing us to do it internally ( I promise to do this ASAP to not block you with your changes ) 

Just prepare your changes and if it's merged before this PR I will have no problems to rebase and handle the conflicts as they arise

Reply:
> I don't understand why `die_on_timeout` is being deprecated and I'd like to learn more about it.
> 
> If we decide to proceed I think that first Oliver's suggestion is enough and will behave the same.

With help of @pdostal  I understand that your suggestion @okurz  is actually valid and will work for us too . I did logical mistake when was analyzing changes ... I am revoking my request changes feel free to merge your suggestion to public cloud part